Adaptors/MongoDB/src/ChangeStreamUpdate.cs

```diff
@@ -0,0 +1,94 @@
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace ArmoniK.Core.Adapters.MongoDB;
+
+public static class ChangeStreamUpdate
+{
+  public static async Task<IChangeStreamCursor<ChangeStreamDocument<T>>> GetUpdates<T>(IMongoCollection<T>                             collection,
+                                                                                       IClientSessionHandle                            sessionHandle,
+                                                                                       Expression<Func<ChangeStreamDocument<T>, bool>> filter,
+                                                                                       IEnumerable<string>                             fields,
+                                                                                       CancellationToken                               cancellationToken = default)
+  {
+    var match2 = new BsonDocument
+                 {
+                   {
+                     "$addFields", new BsonDocument
+                                   {
+                                     {
+                                       "tmpfields", new BsonDocument
+                                                    {
+                                                      {
+                                                        "$objectToArray", "$updateDescription.updatedFields"
+                                                      },
+                                                    }
+                                     },
+                                   }
+                   },
+                 };
+    var match3 = new BsonDocument
+                 {
+                   {
+                     "$match", new BsonDocument
+                               {
+                                 {
+                                   "tmpfields.k", new BsonDocument
+                                                  {
+                                                    {
+                                                      "$in", new BsonArray(fields)
+                                                    },
+                                                  }
+                                 },
+                               }
+                   },
+                 };
```

Comment:
Can surely be transformed into a Linq-style expression

Reply:
I wish it was